### PR TITLE
Created CircuitPython.gitignore

### PR DESCRIPTION
**Pull request by bhavesh-k**
# Pull Request

Thank you for contributing to @toptal/gitignore and https://www.gitignore.io.

## New or update

Select the appropriate check box for this pull request. This helps when merging to ensure there are no conflicts with other templates or misunderstandings of how thee template list works.

### New

- [x] Template - New `.gitignore` template
- [ ] Composition - Template made from smaller templates
- [ ] Inheritance - Template similar to an existing template
- [ ] Patch - Template extending functionality of existing template

### Update

- [ ] Template - Update existing `.gitignore` template

## Details

*replace this section with links and/or info about the proposed request*

### CODE REVIEW

File: templates/CircuitPython.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.Trashes
+.metadata_never_index
+.fseventsd/
```
Comment (filipechagas) [inline templates/CircuitPython.gitignore:3]: `.Trashes` and `.fseventsd` are macOS related. They're being ignored by `macOS.gitignore`. I see that `.metadata_never_index` is also macOS related so you may want to add it to a `macOS.patch` file instead.

**Review by filipechagas — CHANGES_REQUESTED** [review]
Thanks for contributing!

As soon as the requested changes are applied, we'll be happy to accept the PR.

File: templates/CircuitPython.gitignore (hunk 1 above)
Comment (bhavesh-k) [reply templates/CircuitPython.gitignore:3]: @filipechagas Hi, do you need more information on this? I hope my explanation was clear - please let me know if not

Comment (filipechagas) [reply templates/CircuitPython.gitignore:3]: Thank you for giving me context @bhavesh-k ! It makes sense.

**Review by filipechagas — APPROVED** [review]

**Pull request merged**